core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala

```diff
@@ -725,6 +739,17 @@ object DruidBeams
       )
     }
 
+    /**
+      * Provide tunings for communication with Druid tasks. Optional, see [[DruidBeamConfig]] for defaults.
+      *
+      * @param beamConfig beam config tunings
+      * @return new builder
+      */
+    def generalConfig(generalConfig: PropertiesBasedConfig) = {
```

Comment:
This is a bit weird. Generally the Builder is set up to take each config option separately. The idea is that if you're using Tranquility programmatically (i.e. using Core or Storm or Samza), each Builder method call is equivalent to specifying a particular property with a property-based config (i.e. using Server or Kafka).

Any reason it has to be done this way, or would it be possible to split this up into a Builder method for each option?

Reply:
Split this into a builder method for each option